src/sidebar/components/test/top-bar-test.js

```diff
@@ -87,6 +95,33 @@ describe('topBar', function () {
     assert.called(onLogout);
   });
 
+  it("checks whether we're using a third-party service", function () {
+    createTopBar();
+
+    assert.called(fakeIsThirdPartyService);
+    assert.alwaysCalledWithExactly(fakeIsThirdPartyService, fakeSettings);
+  });
+
+  context('when using a first-party service', function () {
+    it('shows the share page button', function () {
+      var el = createTopBar();
+
+      assert.equal(el.find('[title="Share this page"]').length, 1);
```

Comment:
`el` doesn't appear to be a normal DOM element, it's missing lots of the methods, using `length` is the only way I could find to test whether it was found or not.

Might be an idea to add a `js-*` CSS class instead of searching by `title` here. Existing tests use `title` though.

Reply:
Angular in several places wraps DOM elements using a jQuery-like interface called jqLite. See https://docs.angularjs.org/api/ng/function/angular.element.

As with jQuery, functions operate on lists of elements. Hence `length` returns the list size and indexing using `el[0]` returns the actual DOM element.

Reply:
Aha, bookmarked, good to know. I was trying to find some way in the Chrome console to find out what class or type of object this is, but all I could get was "object".

Reply:
Ok, I've fixed this to use `querySelector`